gui/symfony-docker/src/Service/BlockChainService.php

```diff
@@ -135,4 +165,40 @@ public function getStringDataFromTokenID(int $tokenId): array
         return $returnData;
     }
 
+    public function getPossibleResourceFromResourceID(int $resourceId, String $role, String $resourceType): array
+    {
+        $response = $this->httpClient->request('GET', "http://127.0.0.1:8080/resource/templates?required_role=".$role);
+        $data = json_decode($response->getContent(), true);
+        $count = 0;
+        $returnData = [];
+        foreach ($data as $numberOfArray => $datas) {
+            if ($datas["needed_resources"][0] == $resourceId && $datas["resource_type"] == $resourceType) {
+                $returnData[$count++] = $datas;
+            }
+        }
+        // dd($returnData);
+        return $data;
+    }
+
+    public function replaceMetaData($walletAddress,int $tokenId, array $metadata): array
```

Comment:
can you briefly explain why it is called "replace"Metadata and not "add"Metadata ? 
As i can see, we replace nothing but just add metadata, no ?